src/lib/db.js

```diff
@@ -1578,3 +1442,108 @@ export function getSemesterTeams(semesterName, teamIndices) {
     })
   ));
 }
+
+export async function getNumArticles() {
+  const rows = await database('articles').count('* as numArticles');
+  return rows[0].numArticles;
+}
+
+/**
+ * Updates which category an article belongs to
+ * @param {Object} jsonGraphArg - An object of type { [slugs]: articleUpdateObject[] }
+ * @returns {Promise<string[]>} - Slugs of the articles that had their category changed
+ */
+async function updateArticleCategories(jsonGraphArg) {
+  const articleCategoryPairs = _.map(jsonGraphArg, (articleUpdater, slug) => {
+    if (articleUpdater.hasOwnProperty('category')) {
+      return {
+        articleSlug: slug,
+        categorySlug: articleUpdater.category,
+      };
+    }
+    return null;
+  }).filter(x => x);
```

Comment:
This time it is needed haha! It removes all the ones where I returned `null` :)

Reply:
Although, if even you find it confusing to read, maybe I should make it `.filter(x => x !== null)` would that make it clearer? They should be equivalent since it's all objects being returned